### DISCUSSION TITLE
Battery level false alarm [Linux]

### DISCUSSION THREAD

**Discussion by taep96** [ppy/osu#21640] [Q&A] [answered]
### Type

Game behaviour

### Bug description

Low battery alert pops up even when the device doesn't have a battery

### Screenshots or videos

![image](https://user-images.githubusercontent.com/64481039/207651696-d570f685-31a1-4f54-992b-927c5ee3c646.png)

### Version

2022.1208.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/10229778/database.log)
[input.log](https://github.com/ppy/osu/files/10229779/input.log)
[network.log](https://github.com/ppy/osu/files/10229780/network.log)
[performance.log](https://github.com/ppy/osu/files/10229781/performance.log)
[performance-audio.log](https://github.com/ppy/osu/files/10229782/performance-audio.log)
[performance-draw.log](https://github.com/ppy/osu/files/10229783/performance-draw.log)
[performance-input.log](https://github.com/ppy/osu/files/10229784/performance-input.log)
[performance-update.log](https://github.com/ppy/osu/files/10229785/performance-update.log)
[runtime.log](https://github.com/ppy/osu/files/10229786/runtime.log)

**Comment by bdach** [comment, marked as answer]
We rely on SDL2 and [`SDL_GetPowerInfo()`](https://wiki.libsdl.org/SDL2/SDL_GetPowerInfo) for detecting the battery level.

https://github.com/ppy/osu/blob/60c8ef3fe5fe9adcb204814d975b7ed6c832fc8c/osu.Desktop/OsuGameDesktop.cs#L188-L201

So any issues with detection need to be reported to SDL and debugged against it. There is nothing that can be done in this repository to fix it.

I can't reproduce for one, so it's likely something about your system.

**Reply by taep96** [reply]
What's the percentage at which it shows the warning? I think it may be reading my mouse's battery. I charged it and it stopped, now it's at 25% (I think?) and I am getting it again. I will open an issue on SDL repo later too.
![image](https://user-images.githubusercontent.com/64481039/208267541-af89d686-611c-48e1-a8b9-4f3e15968451.png)

**Reply by bdach** [reply]
25%

https://github.com/ppy/osu/blob/b35e796d75153b53c1e450f407a48032aef1eddf/osu.Game/Screens/Play/PlayerLoader.cs#L584

**Reply by peppy** [reply]
Tracking at https://github.com/libsdl-org/SDL/issues/6835 for reference.

**Reply by bdach** [reply]
@peppy upstream [came back with a possible fix for this](https://github.com/libsdl-org/SDL/issues/6835#issuecomment-1553386215), so it may be helpful for us to get a SDL2-CS bump going for verification.

**Reply by peppy** [reply]
https://github.com/ppy/SDL2-CS/actions/runs/5035643254 running a release
